src/array/boolean/mod.rs

```diff
@@ -118,6 +119,16 @@ impl Array for BooleanArray {
     fn slice(&self, offset: usize, length: usize) -> Box<dyn Array> {
         Box::new(self.slice(offset, length))
     }
+    fn with_validity(&self, validity: Option<Bitmap>) -> Result<Box<dyn Array>> {
```

Comment:
What do you think about the following changes:

1. panic instead of `result`, to be aligned with the `from_data` where we also panic if we pass arguments with the wrong length (and also aligned with the usual bound checks from `Rust`)
2. Return `BooleanArray` instead of `Box<dyn Array>`, so that people can keep the typing information if they want to use the typed arrays

Reply:
1. Sure, I'll change that :+1: .
2. I can't. The `Array` trait is implemented for any array implementation, meaning that `with_validity` will return a different implementation of that trait, therefore I wrapped in that trait object.

Reply:
ah, wrt to 2, what about how we are doing for other "trait methods" (such as `clone` and `slice`):

```rust
impl BooleanArray {
     pub fn with_validity(validity: Option<Bitmap>) -> BooleanArray {
          todo!();
     }
}
```

and 

```rust
impl Array {
    fn with_validty(validity: Option<Bitmap>) -> Box<dyn Array> {
        match self.data_type().to_physical_type() {
            PhysicalType::Boolean => Box::new(self.as_any().downcast_ref::<BooleanArray>().unwrap().with_validity(validity))
        }
    }
}
```

Reply:
Oh, right. That makes sense.